### PR TITLE
eth int default

### PR DESCRIPTION
**Pull request by saichint**
This PR for adding purge_config property to interface provider so that the physical interface can be put back into default state. All tests pass

### CODE REVIEW

**Comment by mikewiebe** [comment]
Please reference the GitHub issue number so it gets linked.

**Comment by saichint** [comment]
This update fixes the issue with the cisco_interface provider described in https://github.com/cisco/cisco-network-puppet-module/issues/424

File: lib/cisco_node_utils/interface.rb [hunk 1]
```diff
@@ -1988,5 +1988,17 @@ def set_range_based_params(list, param_name)
         end
       end
     end
+
+    def purge_config=(val)
+      fail ArgumentError, 'Interface MUST be ethernet' unless
```
Comment (mikewiebe) [inline lib/cisco_node_utils/interface.rb]: I would reword this as `'purge_config is only supported on Ethernet interfaces`

File: lib/cisco_node_utils/interface.rb [hunk 2]
```diff
@@ -1988,5 +1988,17 @@ def set_range_based_params(list, param_name)
...
+        @name[/ethernet/]
+      fail ArgumentError, 'Only true is allowed for this config' unless val
```
Comment (mikewiebe) [inline lib/cisco_node_utils/interface.rb]: I would remove the second `fail` and support both `true` and `false`.  You can simply add an early `return unless val` at the beginning of the method.
